main.cpp

```diff
@@ -101,52 +135,116 @@ std::string Piano::displayChoosenSound(double soundKnobValue)
     return organSound;
 }
 
+double Piano::audio(int notePressed, double volume, bool isFootPedalPressed)
+{
+    if (isFootPedalPressed)
+    {
+        return 440.0 * volume;
+    }
+    return notePressed;
+}
+
 
 struct Microwave
 {
-    bool doorIsOpen = true;
-    double secondsRemaining = 0.0;
-    double power = 700.0;
-    int presetDishInput = 0;
+    Microwave();
+    bool doorIsOpen;
+    double secondsRemaining, power;
+    int presetDishInput;
 
     struct Preset
     {
-        float weightInGrams = 500.0f; 
-        int seconds = 1200;  
-        int power = 900; 
-        std::string presetName = "Potato";
+        float weightInGrams; 
+        int seconds;  
+        int power; 
+
+        Preset(float WeightInGrams,int Seconds,int Power);
     };
 
     void cookPotato( Preset potato );
     void cookByWeight(double selectedWeight, double power);
 };
 
+Microwave::Microwave()
+{
+    doorIsOpen = true;
+    secondsRemaining = 0.0;
+    power = 700.0;
+    presetDishInput = 0;
+    std::cout << "Instance of 'Microwave' has been constructed" << std::endl;
+}
+
+Microwave::Preset::Preset(float WeightInGrams,int Seconds,int Power)
+{
```

Comment:
These variable names do not conform the course coding standard:
variable and function names should begin with a lowercase letter
Part 2 instructions in case your names cause warnings:
```
 2) provide implementations for the member functions you declared in your 10 user-defined types from the previous video outside of your UDT definitions.
    If you have 'unused parameter' warnings, you aren't using one of your function parameters in your implementation.
    Solution: use the parameter in your implementation.

    If you have 'shadows a field of <classname>' warnings, a local variable in the function has the same name as a class member.  
    This local variable could also be a function parameter with the same name as the class member.
    Solution: change the local variable's name so it is different from the class member variable's name.
```
- [ ] revise